monitoring/gitlab.py

```diff
@@ -0,0 +1,140 @@
+
+        project, issue = self.get_project_and_issue(gl, issue_subject)
+
+        self.log_debug("Uploading attachments.")
+        issue.uploads = []
+        for item in presentation_list:
+            if item["presentation_type"] == "image":
+                file_name = Path(item["path"]).name
+                upload = project.upload(file_name, filepath=f"{dst_folder}/{file_name}")
+                item["image_link"] = upload[
+                    "markdown"
+                ]  # string of the form "![]()" generated by Gitlab
+
+        # render the template and add as description
+        self.log_debug("Updating the issue description.")
+        issue.description = gitlab_template.render(
+            presentation_list=presentation_list,
+            issue_url=issue.web_url,
+            create_anchor=create_anchor,
+        )
+        self.log_debug("Saving issue.")
+        issue.save()
+
+    def get_presentation_list(self, sources, dst_folder):
+        """create a list of presentation objects"""
+        self.log_debug("Getting list of presentation objects.")
+        presentation_list = []
+        for src in sources:
+            try:
+                try:
+                    pres_object = PresentationObject(dst_folder, **src)
+                except TypeError:
+                    pres_object = PresentationObject(dst_folder, src)
+                self.log_debug(
+                    f"Loading {pres_object.loader.diag_type} diagnostic from {pres_object.loader.path}."
+                )
+                presentation_list.append(pres_object.create_dict())
+            except PresentationException as msg:
+                self.log_warning(f"Can not present diagnostic: {msg}")
+        return presentation_list
+
+    def get_template(self, context, template):
+        """get Jinja2 template file"""
+        search_path = [".", "templates"]
+        if "_se_cmd_cwd" in context:
+            search_path.extend(
+                [
+                    context["_se_cmd_cwd"],
+                    Path(context["_se_cmd_cwd"]) / "templates",
+                ]
+            )
+        self.log_debug(f"Search path for template: {search_path}")
+
+        loader = jinja2.FileSystemLoader(search_path)
+        environment = jinja2.Environment(loader=loader)
+        for name, function in j2filters().items():
+            environment.filters[name] = function
+        return environment.get_template(template)
+
+    def get_project_and_issue(self, gl, issue_subject):
+        """Connect to Gitlab server, find and return correct issue and project based on user input"""
+
+        project_id = 1982
+        try:
+            project = gl.projects.get(project_id)
+        except (
+            gitlab.GitlabAuthenticationError,
+            requests.exceptions.ConnectionError,
+        ) as e:
+            msg = f"Could not log in to Gitlab server ({e})"
+            self.log_error(msg)
+            raise ScriptEngineTaskRunError()
```

Comment:
I do not exactly remember how we did elsewhere, but shouldn't we raise the exception with the message? Or did we discuss this and decided *not* to do it that way?

Reply:
We follow the same approach everywhere else in the package, but I agree that it is strange. I am opening an issue for this as well.
<img width="471" alt="image" src="https://github.com/user-attachments/assets/50440d8a-5667-418b-bce4-19cc465bcbe4" />

Reply:
see #116